### ISSUE TITLE
Issue with anonymous auth and ghcr.io

### ISSUE THREAD

**Issue by imjasonh** [krustlet/oci-distribution#14]
I'm trying to fetch image data about a public image on ghcr.io, using `RegistryAuth::Anonymous`, and it's not working. The same code can successfully fetch from gcr.io and Docker Hub.

I've got a Gist with a minimal repro case: https://gist.github.com/imjasonh/30c95b534487e72afd8f1530072878ed#file-main-rs

tl;dr:

GCR works (so does Docker Hub):

```
$ cargo run gcr.io/imjasonh/test
Digest: sha256:c99359ba4b63a7523525471f82a27fcc887ca2e03ecf6d5847130b2146853c12
```

GHCR doesn't work:

```
$ cargo run ghcr.io/jyotsna-penumaka/test
Error: OCI API error: authentication required on https://ghcr.io/v2/jyotsna-penumaka/test/manifests/latest
```

Both of these images are publicly available and can be fetched with [`crane`](https://github.com/google/go-containerregistry/tree/main/cmd/crane) using `go-containerregistry`.

```
$ crane digest ghcr.io/jyotsna-penumaka/test
sha256:cdcab0563d63cc1f1dee972d685a1d21b8fef98b11c59ce28803f77e6e93a622
```

**Comment by flavio** [comment]
This has been fixed by https://github.com/krustlet/oci-distribution/pull/6 - unfortunately this isn't part of a tagged release yet.

I tried your reproduced and it works fine when using the crate from the `main` branch.

@thomastaylor312, @bacongobbler : I think we should tag a new release of oci-distribution. He have a bunch of improvements merged into the main branch that haven't been pushed out via official release yet.

**Comment by flavio** [comment]
Fixed with the freshly released [v0.8.1](https://github.com/krustlet/oci-distribution/releases/tag/v0.8.1) release.

**Comment by imjasonh** [comment]
> Fixed with the freshly released [v0.8.1](https://github.com/krustlet/oci-distribution/releases/tag/v0.8.1) release.

Excellent! Dependabot beat you to it: https://github.com/sigstore/sget/pull/83 😄 but it was great to see.

Thanks for the fix and for the release!
